### DISCUSSION TITLE
What is the "right" way to use github integration and unique now.json for each branch?

### DISCUSSION THREAD

**Discussion by pejowei** [vercel/vercel#4049] [General] [answered]
Context, using Github integration with Zeit Now.

I would like my workflow to be:
 - code change in dev branch, auto deployed on zeit where I test changes
 - merge dev into master, auto deployed on zeit as production

I want dev and master to use different env vars, thus a modified now.json file. (I'm admittedly a git amateur, so perhaps there is an elegant way to partially merge my dev branch into master I have not yet discovered. Then I could maintain different now.json files in each branch without merging them.)

I am hoping, however, for a way to tell Zeit to use different now.json files for my master and dev deployments so that I can keep my workflow clean, do full merges from dev branch to master and have each branch use different environment variables.

Couldn't find a prior question or doc on this -- can someone point me in the right direction?

**Comment by AndrewBarba** [comment, marked as answer]
Check out: https://github.com/zeit/now/discussions/4048

We're almost there...

**Reply by pejowei** [reply]
Awesome, that would solve my problem.  Thanks @AndrewBarba !

**Comment by Vadorequest** [comment]
@pejowei Check out https://github.com/UnlyEd/next-right-now and see how we did it. (github actions, with the workflow you describe)
